subsys/bluetooth/mesh/rpl_storage.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include <bluetooth/mesh.h>
+
+#include "net.h"
+#include "rpl.h"
+
+/* Covers no persistent storage usecase */
+static struct bt_mesh_rpl replay_list[CONFIG_BT_MESH_CRPL];
+
+void bt_mesh_rpl_strg_store(struct bt_mesh_rpl *rpl, bool force)
+{
+}
+
+void bt_mesh_rpl_strg_clear(struct bt_mesh_rpl *rpl)
+{
+	(void)memset(rpl, 0, sizeof(*rpl));
+}
+
+void bt_mesh_rpl_strg_clear_all(void)
+{
+	(void)memset(replay_list, 0, sizeof(replay_list));
+}
+
+struct bt_mesh_rpl *bt_mesh_rpl_strg_iter(struct bt_mesh_rpl *rpl)
+{
+	if (!rpl) {
+		return &replay_list[0];
+	}
+
+	int idx = rpl - replay_list;
```

Comment:
What about adding "static inline int rpl_idx(const struct bt_mesh_rpl *rpl)", so this functions look the same as in rpl_strg_settings.c.